step_command.go

```diff
@@ -26,6 +26,12 @@ type Signature struct {
 //
 // Standard caveats apply - see the package comment.
 type CommandStep struct {
+	// Fields common to various step types
+	Key       string   `yaml:"key,omitempty" aliases:"id,identifier"`
+	Name      string   `yaml:"name,omitempty" aliases:"label"`
```

Comment:
i was about to add a comment saying "i feel like `label` is the "more default" name", but it turns out we mix and match `label` and `name` in this very repo: 
![CleanShot 2023-11-22 at 17 51 07@2x](https://github.com/buildkite/go-pipeline/assets/15753101/6a469f92-bbd3-4ba7-bb63-ef285cee8b70)

Reply:
On the one hand, the docs talk about `label` primarily and don't even mention `name` as an alias. On the other hand we do this in the backend: `name: config["name"] || config["label"],`

Reply:
I changed it.

Reply:
i reckon if the docs talk about label, that should be the default - the stuff in the database doesn't really matter to users of this library